velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -1511,6 +1512,39 @@ INSTANTIATE_TEST_SUITE_P(
         common::CompressionKind::CompressionKind_LZ4,
         common::CompressionKind::CompressionKind_GZIP));
 
+class Foo {
+ public:
+  explicit Foo(int64_t id) : id_(id) {}
+
+  int64_t id() const {
+    return id_;
+  }
+
+  static std::shared_ptr<Foo> create(int64_t id) {
+    // Return the same instance if the id is already in the map, to make
+    // the operator== work with the instance before and after serde.
```

Comment:
not sure I followed. Why do you need to compare that they are the same instances, and not just different instances which are equal?

Reply:
I think it's because the comparison of shared_ptr<Foo> will compare the raw pointer address, so it needs to be the same instance. we could change the test utilities to de-reference before comparisons if desired.